0.11.3 CLI with 0.11.2 NPM packages

Is it expected that current `pulumi` CLI working with older NPM packages will fail like this?  Or is this some problem with my local machine setup?

```
$ pulumi preview
Previewing stack 'kubernetes-ts-guestbook-luke' in the Pulumi Cloud ☁️
global: * Unchanged, 1 warning(s). warning: resource plugin kubernetes is expected to have version >=0.11.2, but has 0.11.2-1523383521-g8e03a90-dirty; the wrong version may be on your path, or this may be a bug in the plugin
Previewing changes:

global: * Would not change, 15 error(s), 1 warning(s). info:     at Object.<anonymous> (/Users/luke/go/src/github.com/pulumi/examples/kubernetes-ts-guestbook/node_modules/@pulumi/pulumi/runtime/closure/serializeClosure.js:12:17)

global: Diagnostics
  warning: resource plugin kubernetes is expected to have version >=0.11.2, but has 0.11.2-1523383521-g8e03a90-dirty; the wrong version may be on your path, or this may be a bug in the plugin

  info: /Users/luke/go/src/github.com/pulumi/examples/kubernetes-ts-guestbook/node_modules/@pulumi/pulumi/runtime/closure/createClosure.js:38

  info:         throw new errors_1.RunError(`Failed to locate custom Pulumi SDK Node.js extension. This is a bug! (${err.message})`);

  info:         ^

  info:

  info: Error: Failed to locate custom Pulumi SDK Node.js extension. This is a bug! (Cannot find module 'nativeruntime-v0.11.0.node')

  info:     at RunError (/Users/luke/go/src/github.com/pulumi/examples/kubernetes-ts-guestbook/node_modules/@pulumi/pulumi/errors.js:10:9)

  info:     at Object.<anonymous> (/Users/luke/go/src/github.com/pulumi/examples/kubernetes-ts-guestbook/node_modules/@pulumi/pulumi/runtime/closure/createClosure.js:38:15)

  info:     at Module._compile (module.js:570:32)

  info:     at Object.Module._extensions..js (module.js:579:10)

  info:     at Module.load (module.js:487:32)

  info:     at tryModuleLoad (module.js:446:12)

  info:     at Function.Module._load (module.js:438:3)

  info:     at Module.require (module.js:497:17)

  info:     at require (internal/module.js:20:19)

  info:     at Object.<anonymous> (/Users/luke/go/src/github.com/pulumi/examples/kubernetes-ts-guestbook/node_modules/@pulumi/pulumi/runtime/closure/serializeClosure.js:12:17)

error: an error occurred while advancing the preview
```

--- a/build.proj
+++ b/build.proj
@@ -20,6 +20,13 @@
           WorkingDirectory="$(NodeJSSdkDirectory)" />
   </Target>
 
+  <Target Name="EnsurePrebuilt">
+    <MakeDir Directories="$(NodeJSSdkDirectory)\prebuilt" />
+    <Exec Command="aws s3 cp s3://eng.pulumi.com/nativeruntime/windows/nativeruntime.node $(NodeJSSdkDirectory)\prebuilt\nativeruntime.node" />
+    <Exec Command="aws s3 cp s3://eng.pulumi.com/nativeruntime/windows/nativeruntime-v0.11.0.node $(NodeJSSdkDirectory)\prebuilt\nativeruntime-v0.11.0.node" />
+    <Exec Command="aws s3 cp s3://eng.pulumi.com/nativeruntime/windows/pulumi-language-nodejs-node.exe $(NodeJSSdkDirectory)\prebuilt\pulumi-language-nodejs-node.exe" />
+  </Target>
+
   <Target Name="TypeScriptCompileNodeSdk">
     <Exec Command="&quot;$(MSBuildThisFileDirectory)\scripts\get-version.cmd&quot;" ConsoleToMSBuild="true" Condition="'$(Version)' == ''">
       <Output TaskParameter="ConsoleOutput" PropertyName="Version" />
@@ -70,6 +77,11 @@
           DependsOnTargets="BinPlaceNodeSdkProtos;BinPlaceNodeSdkTestData;YarnLinkSdk">
      <Copy SourceFiles="$(NodeJSSdkDirectory)\dist\pulumi-language-nodejs-exec.cmd" DestinationFolder="$(PulumiBin)" />
      <Copy SourceFiles="$(NodeJSSdkDirectory)\dist\pulumi-resource-pulumi-nodejs.cmd" DestinationFolder="$(PulumiBin)" />
+     <MakeDir Directories="$(PulumiBin)\v6.10.2" />
+     <MakeDir Directories="$(PulumiBin)\custom_node" />
+     <Copy SourceFiles="$(NodeJSSdkDirectory)\prebuilt\nativeruntime.node" DestinationFolder="$(PulumiBin)\v6.10.2" />
+     <Copy SourceFiles="$(NodeJSSdkDirectory)\prebuilt\nativeruntime-v0.11.0.node" DestinationFolder="$(PulumiBin)\v6.10.2" />
+     <Copy SourceFiles="$(NodeJSSdkDirectory)\prebuilt\pulumi-language-nodejs-node.exe" DestinationFiles="$(PulumiBin)\custom_node\node.exe" />
   </Target>
 
   <Target Name="BuildNodeSdk"
@@ -90,7 +102,7 @@
   </Target>
 
   <Target Name="Build"
-          DependsOnTargets="EnsureGoDependencies;EnsureNodeDependencies;BuildNodeSdk;BuildGoCmds">
+          DependsOnTargets="EnsureGoDependencies;EnsureNodeDependencies;EnsurePrebuilt;BuildNodeSdk;BuildGoCmds">
   </Target>
 
   <Target Name="IntegrationTest">

--- a/scripts/make_release.ps1
+++ b/scripts/make_release.ps1
@@ -35,6 +35,12 @@ CopyPackage "$Root\sdk\nodejs\bin" "pulumi"
 Copy-Item "$Root\sdk\nodejs\dist\pulumi-language-nodejs-exec.cmd" "$PublishDir\bin"
 Copy-Item "$Root\sdk\nodejs\dist\pulumi-resource-pulumi-nodejs.cmd" "$PublishDir\bin"
 
+New-Item -ItemType Directory -Path "$PublishDir\bin\v6.10.2"
+New-Item -ItemType Directory -Path "$PublishDir\bin\custom_node"
+Copy-Item "$Root\sdk\nodejs\prebuilt\nativeruntime.node" "$PublishDir\bin\v6.10.2"
+Copy-Item "$Root\sdk\nodejs\prebuilt\nativeruntime-v0.11.0.node" "$PublishDir\bin\v6.10.2"
+Copy-Item "$Root\sdk\nodejs\prebuilt\pulumi-language-nodejs-node.exe" "$PublishDir\bin\custom_node\node.exe"
+
 # By default, if the archive already exists, 7zip will just add files to it, so blow away the existing
 # archive if it exists.
 if (Test-Path $PublishFile) {

--- a/scripts/make_release.sh
+++ b/scripts/make_release.sh
@@ -56,6 +56,10 @@ cp ${ROOT}/sdk/python/cmd/pulumi-language-python-exec ${PUBDIR}/bin/
 # Copy packages
 copy_package "${ROOT}/sdk/nodejs/bin/." "@pulumi/pulumi"
 
+# Copy prebuilt native modules
+mkdir ${PUBDIR}/bin/v6.10.2/
+cp ${ROOT}/sdk/nodejs/prebuilt/*.node ${PUBDIR}/bin/v6.10.2/
+
 # Tar up the file and then print it out for use by the caller or script.
 tar -czf ${PUBFILE} -C ${PUBDIR} .
 echo ${PUBFILE} ${PUBTARGETS[@]}

--- a/sdk/nodejs/.gitignore
+++ b/sdk/nodejs/.gitignore
@@ -1,5 +1,6 @@
 /bin/
 /coverage/
+/prebuilt/
 /node_modules/
 /custom_node/
 /runtime/native/node_dev/

--- a/sdk/nodejs/Makefile
+++ b/sdk/nodejs/Makefile
@@ -15,6 +15,11 @@ include ../../build/common.mk
 
 export PATH:=$(shell yarn bin 2>/dev/null):$(PATH)
 
+ensure::
+	[ -d "./prebuilt/" ] || mkdir "./prebuilt"
+	[ -e "./prebuilt/nativeruntime.node" ] || aws s3 cp "s3://eng.pulumi.com/nativeruntime/$$(go env GOOS)/nativeruntime.node" "./prebuilt/"
+	[ -e "./prebuilt/nativeruntime-v0.11.0.node" ] || aws s3 cp "s3://eng.pulumi.com/nativeruntime/$$(go env GOOS)/nativeruntime-v0.11.0.node" "./prebuilt/"
+
 lint::
 	$(GOMETALINTER) cmd/pulumi-language-nodejs/main.go | sort ; exit "$${PIPESTATUS[0]}"
 	tslint -c tslint.json -p tsconfig.json
@@ -33,6 +38,8 @@ install::
 	GOBIN=$(PULUMI_BIN) go install -ldflags "-X github.com/pulumi/pulumi/pkg/version.Version=${VERSION}" ${LANGUAGE_HOST}
 	cp dist/pulumi-language-nodejs-exec "$(PULUMI_BIN)"
 	cp dist/pulumi-resource-pulumi-nodejs "$(PULUMI_BIN)"
+	mkdir -p "$(PULUMI_BIN)/v6.10.2"
+	cp ./prebuilt/*.node "$(PULUMI_BIN)/v6.10.2/"
 	rm -rf "$(PULUMI_NODE_MODULES)/$(NODE_MODULE_NAME)/tests"
 
 test_fast::

